src/main/java/org/opensearch/knn/index/codec/VectorAsArraySerializer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.codec;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+/**
+ * Class implements VectorSerializer based on standard Java serialization/deserialization as a single array object
+ */
+public class VectorAsArraySerializer implements VectorSerializer {
```

Comment:
Should we call KNNVector instead of Vector, since Vector means something else in java, just to avoid some confusion. What do you think?

Reply:
I don't have a strong preference as I see your point and agree, at the same time the existing field called Vector, so I was thinking this name follows the convention. 
If you've suggesting KNNVector I assume that "Vector" is not following the convention, so I'll rename my class to KNNVector it in next revision.

Reply:
made the change, resolving